### PR TITLE
Add support for models with dynamic shapes

### PR DESCRIPTION
**Pull request by atobiszei**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -136,8 +137,11 @@ and
 [`sequence_batching`](https://github.com/triton-inference-server/server/blob/main/docs/user_guide/model_configuration.md#sequence-batcher)
 is provided, then `dynamic_batching` will be enabled with default settings.
 
+It is recommended to use OpenVINO dynamic batch size instead.
```
Comment (dtrawins) [inline README.md:140]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -202,11 +203,37 @@ model_repository/
 
 ```
```
Comment (dtrawins) [inline README.md:205]:
```suggestion
Other allowed values are `model.pdmodel` or `model.onnx`.
```

File: README.md [hunk 3]
```diff
@@ -202,11 +203,37 @@ model_repository/
...
 
+### Reshaping model to support dynamic batch size or dimension
```
Comment (dtrawins) [inline README.md:206]:
```suggestion
### Reshaping models
```

File: README.md [hunk 4]
```diff
@@ -202,11 +203,37 @@ model_repository/
...
+
+*Note*: If model is originally exported with dynamic shapes support there is no need to manually specify dynamic shapes in config.
 
-* Models with dynamic shape are not supported in this backend now.
+```
+input [
+  {
+    name: "input"
+    data_type: TYPE_FP32
+    dims: [ -1, 3, -1, 224]
```
Comment (dtrawins) [inline README.md:218]:
```suggestion
    dims: [ -1, 3, -1, -1]
```

File: README.md [hunk 5]
```diff
@@ -202,11 +203,37 @@ model_repository/
 
 ```
 
+### Reshaping model to support dynamic batch size or dimension
 
-## Known Issues
+Following section shows how to use OpenVINO dynamic shapes. `-1` denotes dimension accepting any value on input. In this case
+while model originally accepted input with layout `NCHW` and shape `(1,3,224,224)`, now it accepts any batch size and height.
+
+*Note*: If model is originally exported with dynamic shapes support there is no need to manually specify dynamic shapes in config.
```
Comment (dtrawins) [inline README.md:211]:
```suggestion
*Note*: If the model is originally exported with dynamic shapes, there is no need to manually specify dynamic shapes in config.
```

File: README.md [hunk 6]
```diff
@@ -202,11 +203,37 @@ model_repository/
...
+string_value:"yes"
+}
+}
+```
 
-* As of now, the Openvino backend does not support variable shaped tensors. However, the dynamic batch sizes in the model are supported. See `SKIP_OV_DYNAMIC_BATCHSIZE` and `ENABLE_BATCH_PADDING` parameters for more details.
+## Known Issues
 
 * Models with the scalar on the input (shape without any dimension are not supported)
+* Reshaping using dimension [ranges](https://docs.openvino.ai/2023.3/ovms_docs_dynamic_shape_dynamic_model.html) is not supported.
```
Comment (dtrawins) [inline README.md:239]:
```suggestion
* Reshaping using [dimension ranges](https://docs.openvino.ai/2023.3/ovms_docs_dynamic_shape_dynamic_model.html) is not supported.
```

File: README.md (hunk 4 above)
Comment (atobiszei) [reply README.md:218]: This is now inconsistent with comment in L209

File: README.md [hunk 7]
```diff
@@ -201,12 +201,38 @@ model_repository/
     └── config.pbtxt
 
 ```
+Other allowed values are `model.pdmodel` or `model.onnx`.
+### Reshaping models
 
+Following section shows how to use OpenVINO dynamic shapes. `-1` denotes dimension accepting any value on input. In this case
+while model originally accepted input with layout `NCHW` and shape `(1,3,224,224)`, now it accepts any batch size and height.
```
Comment (dtrawins) [inline README.md:208]:
```suggestion
while model originally accepted input with layout `NCHW` and shape `(1,3,224,224)`, now it accepts any batch size and resolution.
```

**Review by dtrawins — APPROVED** [review]

**Pull request merged**